scene/animation/animation_blend_tree.cpp

```diff
@@ -1328,6 +1328,24 @@ void AnimationNodeBlendTree::get_node_connections(List<NodeConnection> *r_connec
 			}
 		}
 	}
+	// Keep node connections sorted for deterministic scene file output.
+	struct Comparator {
+		bool operator()(const NodeConnection &a, const NodeConnection &b) const {
+			String a_input_node(a.input_node);
+			String b_input_node(b.input_node);
+			if (a_input_node < b_input_node) {
+				return true;
+			} else if (a_input_node == b_input_node) {
```

Comment:
Better to compare the original `StringName`, much faster

Reply:
I'd also suggest using `StringName::AlphCompare`

Reply:
awesome, that's more efficient than the temporary strings and seems to work fine as well, thanks! :) will update 3.x PR momentarily as well.